cmd/integration/integration.go

```diff
@@ -373,55 +376,55 @@ func runReplicationControllerTest(c *client.Client) {
 
 func runAPIVersionsTest(c *client.Client) {
 	v, err := c.ServerAPIVersions()
+	clientVersion := c.APIVersion()
 	if err != nil {
 		glog.Fatalf("failed to get api versions: %v", err)
 	}
-	if e, a := []string{"v1beta1", "v1beta2"}, v.Versions; !reflect.DeepEqual(e, a) {
-		glog.Fatalf("Expected version list '%v', got '%v'", e, a)
+	// Verify that the server supports the API version used by the client.
+	for _, version := range v.Versions {
+		if version == clientVersion {
+			glog.Infof("Version test passed")
+			return
+		}
 	}
-	glog.Infof("Version test passed")
+	glog.Fatalf("Server does not support APIVersion used by client. Server supported APIVersions: '%v', client APIVersion: '%v'", v.Versions, clientVersion)
 }
 
 func runSelfLinkTestOnNamespace(c *client.Client, namespace string) {
-	var svc api.Service
-	err := c.Post().
-		NamespaceIfScoped(namespace, len(namespace) > 0).
-		Resource("services").Body(
-		&api.Service{
-			ObjectMeta: api.ObjectMeta{
-				Name:      "selflinktest",
-				Namespace: namespace,
-				Labels: map[string]string{
-					"name": "selflinktest",
-				},
+	svcBody := api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name:      "selflinktest",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"name": "selflinktest",
 			},
-			Spec: api.ServiceSpec{
-				Port: 12345,
-				// This is here because validation requires it.
-				Selector: map[string]string{
-					"foo": "bar",
-				},
-				Protocol:        "TCP",
-				SessionAffinity: "None",
+		},
+		Spec: api.ServiceSpec{
+			Port: 12345,
+			// This is here because validation requires it.
+			Selector: map[string]string{
+				"foo": "bar",
 			},
+			Protocol:        "TCP",
+			SessionAffinity: "None",
 		},
-	).Do().Into(&svc)
+	}
+	services := c.Services(namespace)
+	svc, err := services.Create(&svcBody)
 	if err != nil {
 		glog.Fatalf("Failed creating selflinktest service: %v", err)
 	}
-	// TODO: this is not namespace aware
-	err = c.Get().RequestURI(svc.SelfLink).Do().Into(&svc)
+	err = c.Get().RequestURI(svc.SelfLink).Do().Into(svc)
 	if err != nil {
 		glog.Fatalf("Failed listing service with supplied self link '%v': %v", svc.SelfLink, err)
 	}
 
-	var svcList api.ServiceList
-	err = c.Get().NamespaceIfScoped(namespace, len(namespace) > 0).Resource("services").Do().Into(&svcList)
+	svcList, err := services.List(labels.Everything())
```

Comment:
It looks like its still there, but services is now ServiceNamespacer that has already been scoped to the namespace in question.